lua/amp/server/init.lua

```diff
@@ -51,6 +51,11 @@ function M.start(auth_token)
 			else
 				logger.debug("server", "IDE client connected (no auth):", client.id)
 			end
+
+			-- Send initial state to newly connected client
+			vim.defer_fn(function()
+				M._send_state()
+			end, 50)
```

Comment:
why the delay?

Reply:
It seems like the Amp CLI isn't ready to receive information immediately after connecting. I haven't looked into it from that side yet, might be the order in which it sends a connection request and sets up state to track info from the IDE